### DISCUSSION TITLE
Row Level Security (or Trigger) only specific column

### DISCUSSION THREAD

**Discussion by Alvaro203204** [supabase/supabase#16881] [Questions]
Hello, I'm doing and app and i need allow any user authenticated pop or push her own "auth.uid()" to the column "likes" of a row of the "posts" table.

I have tried something with RLS but doesn't work and the max approach was with trigger executing this function before update.
```sql
BEGIN 
 IF NEW.likes IS DISTINCT FROM OLD.likes THEN
    IF ARRAY(SELECT DISTINCT unnest(NEW.likes) EXCEPT SELECT DISTINCT unnest(OLD.likes)) = ARRAY[auth.uid()] OR
    ARRAY(SELECT DISTINCT unnest(OLD.likes) EXCEPT SELECT DISTINCT unnest(NEW.likes)) = ARRAY[auth.uid()] THEN
      RETURN NEW;
    ELSE
      RAISE EXCEPTION 'Likes values must be distinct except for auth.uid()';
    END IF;
  END IF;
  RETURN NEW;
END;
```

This code works but I want to make that the Supabase Dashboard will bypass this restriction, I searched through the Docs and asked to the SupabaseAI but I couldn't find anything that bypass that or any way to make it with RLS.

Any suggestions? Thx in advance 💙

**Comment by GaryAustin1** [comment]
This was solved in Supabase Discord.   Maybe the OP will post the final code they used.
